File: app/src/main/java/com/arturogutierrez/openticator/application/OpenticatorApplication.java
package com.arturogutierrez.openticator.application;

import android.app.Application;
import com.arturogutierrez.openticator.di.component.ApplicationComponent;
import com.arturogutierrez.openticator.di.component.DaggerApplicationComponent;
import com.arturogutierrez.openticator.di.module.ApplicationModule;
import com.karumi.dexter.Dexter;

public class OpenticatorApplication extends Application {

  private ApplicationComponent applicationComponent;

  @Override
  public void onCreate() {
    super.onCreate();

    initialize();
  }

  public ApplicationComponent getApplicationComponent() {
    return applicationComponent;
  }

  private void initialize() {
    initializeDependencyInjector();
    initializeDexter();
  }

  private void initializeDependencyInjector() {
    applicationComponent =
        DaggerApplicationComponent.builder().applicationModule(new ApplicationModule(this)).build();
  }

  private void initializeDexter() {
    Dexter.initialize(this);
  }
}

File: app/src/main/java/com/arturogutierrez/openticator/application/OpenticatorApplication.kt
package com.arturogutierrez.openticator.application

import android.app.Application
import com.arturogutierrez.openticator.di.component.ApplicationComponent
import com.arturogutierrez.openticator.di.component.DaggerApplicationComponent
import com.arturogutierrez.openticator.di.module.ApplicationModule
import com.karumi.dexter.Dexter

class OpenticatorApplication : Application() {

  lateinit var applicationComponent: ApplicationComponent
    private set

  override fun onCreate() {
    super.onCreate()
    initialize()
  }

  private fun initialize() {
    initializeDependencyInjector()
    initializeDexter()
  }

  private fun initializeDependencyInjector() {
    applicationComponent = DaggerApplicationComponent.builder().applicationModule(ApplicationModule(this)).build()
  }

  private fun initializeDexter() {
    Dexter.initialize(this)
  }
}
